site/src/pages/release-notes/1.18.0.mdx

```diff
@@ -0,0 +1,134 @@
+formatted documentation for annotated services #4299 #4304
+
+```java Markdown
+@Description(value = "### Markdown format", markup = Markup.MARKDOWN)
+@Get("/markdown")
+public HttpResponse markdown() {
+    return HttpResponse.of(200);
+}
+```
+
+```java Mermaid
+@Description(value = "graph TD;\n" +
+                     " A-->B;\n" +
+                     " A-->C;\n" +
+                     " B-->D;\n" +
+                     " C-->D;",
+             markup = Markup.MERMAID)
+@Get("/mermaid")
+public HttpResponse mermaid() {
+    return HttpResponse.of(200);
+}
+```
+
+- It is now possible to bind decorators to gRPC services using <type://GrpcServiceBuilder>. #4311 #4316
+```java
+GrpcService.builder()
+           .addService(myGrpcService, List.of(LoggingService.newDecorator()))
+           .build();
+```
+
+- Armeria's gRPC users can now use an <type://UnframedGrpcErrorHandler> which provides a
+[richer error model](https://cloud.google.com/apis/design/errors#error_model) for unframed requests.
+
+```java
+Server.builder()
+      .service(GrpcService.builder()
+                          .addService(grpcService)
+                          .unframedGrpcErrorHandler(UnframedGrpcErrorHandler.ofRichJson())
+                          .build());
+```
+
+```json
+{
+	"error": {
+		"code": 2,
+		"message": "A custom message!"
+	}
+}
+```
+
+- Multipart file upload is now available for GraphQL users. #3971 #4199
+```java
+GraphqlService service =
+    GraphqlService.builder()
+...
+                  .runtimeWiring(c -> {
+                      c.scalar(MoreScalars.multipartFile());
+                      DataFetcher<String> upload = environment -> {
+                          MultipartFile multipartFile = environment.getArgument("file");
+                          ...
+                      }
+                      c.type("Multipart", typeWiring -> typeWiring.dataFetcher("upload", upload));
+                  })
+...
+```
+
+- Users can now extract the innermost element of <type://Unwrappable> type objects easily using <type://Unwrappable#unwrapAll()> #4338
+```java
+FooDecorator extends SimpleDecoratingHttpClient {
+...
+    @Override
+    public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
+        Object innermost = unwrapAll();
+        ...
+    }
+}
+```
+
+## 📈 Improvements
+
+- <type://AnnotatedServiceBindingBuilder> and <type://VirtualHostAnnotatedServiceBindingBuilder> now
+extend a single interface. #4205 #4210
+- <type://RequestContext> now extends <type://Unwrappable>. #4308
+
+## 🛠️ Bug fixes
+
+- You no longer see an intermittent <type://IllegalStateException> for gRPC services
+when <type://GrpcServiceBuilder#useBlockingTaskExecutor(boolean)> is enabled. #4367 #4374
+- Requests no longer hang when gRPC-JSON transcoding is enabled and gRPC decorators return an unframed response. #4337 #4348
+- Files from OSGI bundle classloader are correctly served #4371 #4373
```

Comment:
It seems that `class loader` is officially used.
https://docs.oracle.com/javase/7/docs/api/java/lang/ClassLoader.html
```suggestion
- Files from OSGI bundle class loader are now correctly served #4371 #4373
```